core/lib/spree/testing_support/dummy_app.rb

```diff
@@ -17,6 +17,7 @@
 
 # @private
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
```

Comment:
Do you think this is necessary?

Reply:
I tried to not include that line, but I ran into an issue in `frontend/app/controllers/spree/orders_controller.rb` on line 14: `skip_before_action :verify_authenticity_token, only: [:populate]`

The error was: 
`ArgumentError: Before process_action callback :verify_authenticity_token has not been defined`

I think the problem was that `:verify_authenticity_token` doesn't exist, so it's raising an exception when trying to skip it. My first thought was just to add `raise: false` like this:

`skip_before_action :verify_authenticity_token, only: [:populate], raise: false`

Which works, but seems really hacky and not ideal. Instead, I thought if I added the line:

`protect_from_forgery with: :exception`

It would add `verify_authenticity_token` to the `before_filter` list, which can then be skipped in other controllers without an error.

It's possible I'm misunderstanding, so please let me know what you think.

Thanks so much

Reply:
Does this error happen in the test suite? If yes can you please point a spec that is failing for reference?

Reply:
This is the error output:
```
An error occurred while loading ./spec/controllers/spree/orders_controller_ability_spec.rb.
Failure/Error: skip_before_action :verify_authenticity_token, only: [:populate]

ArgumentError:
  Before process_action callback :verify_authenticity_token has not been defined
# ./app/controllers/spree/orders_controller.rb:14:in `<class:OrdersController>'
# ./app/controllers/spree/orders_controller.rb:4:in `<module:Spree>'
# ./app/controllers/spree/orders_controller.rb:3:in `<top (required)>'
# ./spec/controllers/spree/orders_controller_ability_spec.rb:6:in `<module:Spree>'
# ./spec/controllers/spree/orders_controller_ability_spec.rb:5:in `<top (required)>'


Finished in 0.00005 seconds (files took 4.88 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples

```
It looks like the error is happening when loading the order controller before the tests.

Reply:
Oh ok, it makes sense. I think that we need that `protect_from_forgery with: :exception` then. 
If we add `, raise: false` it will also change the behavior outside the test environment so I don't think it's a viable solution.

Reply:
Yeah good point about keeping it in the test environment. Thanks for taking a look